File: 0974-subarray-sums-divisible-by-k/0974-subarray-sums-divisible-by-k.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var subarraysDivByK = function(nums, k) {
    let ans = 0, pre = 0;
    let mods = new Array(k).fill(0);
    mods[0] = 1;
    
    for (let i of nums)
    {
        pre = (pre + i % k + k) % k;
        ans += mods[pre];
        ++mods[pre];
    }
    return ans;
};

File: 0974-subarray-sums-divisible-by-k/0974-subarray-sums-divisible-by-k.cpp
class Solution {
public:
    int subarraysDivByK(vector<int>& nums, int k) {
        int ans = 0, pre = 0;
        vector<int> mod(k);
        mod[0] = 1;
        
        for (auto &i : nums)
        {
            pre = (pre + i % k + k) % k;
            ans += mod[pre];
            ++mod[pre];
        }
        return ans;
    }
};